finagle-core/src/main/scala/com/twitter/finagle/tracing/Trace.scala

```diff
@@ -70,6 +72,8 @@ object Trace {
       val trace64 = ByteArrays.get64be(bytes, 16)
       val flags64 = ByteArrays.get64be(bytes, 24)
 
+      val traceIdHigh = if(body.length == 40) ByteArrays.get64be(bytes, 32) else 0L
```

Comment:
See comment above in Id.scala.

Reply:
My reply to your referenced comment is kinda hidden now. I think this was the last PR feedback, so I wanted to follow up.

If 0L isn't converted to None via the getter in `Trace`, this will allow propagation of invalid 128 bit TraceIDs (such as a stringified long which is left-padded with 0s). The getter protects against this, with the effect of reducing the span to 64 bit.  Is that not acceptable, or do you think it should propagate ids with invalid high bits?

I hadn't changed this because the guard on 0L works similarly to the previous implementation which reduce all 128bit TraceIDs to 64bit, and I was waiting for a response to my previous comment before changing.

/cc @adriancole